#159 Code sanitisation (Updated)

LeoLuosifen opened the pull request:
## Code Sanitisation #159
- Created a global column names python file to store the columns, such as YEAR = "year", MONTH = "month", etc. These colunm names are considered constant, so use uppercase words.
- Created a global element ids python file to store id variables in different files, and when a variable appears in different files, store the variable as ID_filename_variable value, such as ID_T_RH_DF_STORE = "df-store", ID_EXPLORER_DF_STORE = "df-store".
- Replaced all of these values that using these two files. New variable names can be added in these two files, which allows for expansion and prevents hard coding.

## Contributing.md
- Add the content of Fork & branch processing and Pull Request Regulation, updated the content of Submitting changes, that can guide developers who want to improve or contribute to this project.

We are not sure if these changes meet your needs. If there are any parts that need improvement, please comment and we will check and make changes in time.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Navigation styling refreshed (pill-style, justified tabs); minor UI text refinements.

- Bug Fixes
  - More stable solar-radiation summaries (divide-by-zero guarded).
  - Terminology corrected (psychrometric) and multiple small formatting fixes reducing UI/document glitches.

- Documentation
  - Expanded contributing guide (workflow, tooling, Python 3.11); added/updated videos, images, links, and clarity improvements.

- Refactor
  - Improved UI consistency and stability via centralized identifiers and standardized data naming.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

LeoLuosifen committed:
docs: Add the content of Fork & branch processing

Sallie00 committed:
Add Code style and Testing

ziqi0067 committed:
docs: Add the "Pull Request Regulation" part.

ziqi0067 committed:
docs: Fixed the "pip" to "pipenv".

LeoLuosifen committed:
docs: Fixed the content of the Pull Request Regulation format

FengW01 committed:
docs(contributing): update introduction section of How to contribute guide

FengW01 committed:
Merge remote-tracking branch 'origin/yluo0664' into fwan

FengW01 committed:
docs(contributing): update introduction section of How to contribute guide

FengW01 committed:
docs: update contributing guide

ziqi0067 committed:
docs: A strategy module has been added to the Pull Request section, along with examples of code modifications.

LeoLuosifen committed:
feat: add Global Column Names class to replace string

Commit:
code sanisitation(First version)

LeoLuosifen committed:
Merge branch 'ltc' into yluo0664

LeoLuosifen committed:
fix: updated the comment

Commit:
code sanisitation(Version 2)

LeoLuosifen committed:
fix: replaced the string that using Class ColNames

LeoLuosifen committed:
fix: replaced the string that using Class ColNames in natural_ventilation.py

LeoLuosifen committed:
fix: replaced the string that using Class ColNames in extract_df.py and template_graphs.py

Sallie00 committed:
docs: update pull request regulation of contributing.md

Sallie00 committed:
fix: replaced the string that using Class ColNames in wind.py and charts_data_explorer.py

LeoLuosifen committed:
fix: replaced the string that using Class ColNames in charts_sun.py

Sallie00 committed:
fix: replaced the string that using Class ColNames in utils.py and psy-chart.py and summary.py

Kira-Liu00 committed:
fix: Updated ElementIds  of global_column_names.py

Sallie00 committed:
fix: delete components.py

ziqi0067 committed:
Fix:Updated the ElementIDs of the explorer.py and global_column_names.py

Kira-Liu00 committed:
fix: Updated ElementIds of the psy-chart  and  global _column _names.py

Kira-Liu00 committed:
Merge origin/Ziqi into Kira_Liu: take theirs for global_column_names.py

Kira-Liu00 committed:
Fix: Updated the ElementIds of global_column_names.py

ziqi0067 committed:
Fix:Updated the ElementIDs of the explorer.py, global_column_names.py and natural_ventilation.py.

Kira-Liu00 committed:
Merge remote-tracking branch 'origin/Ziqi' into Kira_Liu

Kira-Liu00 committed:
fix: Updated the ElementIds of outdoor.py, psy-chart.py, global_column_names.py and natural_ventilation.py

FengW01 committed:
Fix: update the sun.py, wind.py and global_column_names.py

Commit:
Fix:
1.add a new python file "global_elementids.py" and move the class ElementIds to this file
2.updated the ElementIDs of the explorer.py,global_column_names.py,natural_ventilation.py,outdoor.py,psy-chart.py,select.py,summary.py,sun.py,t_rh.py,wind.py

Commit:
Fix:Align development branch versions to avoid conflicts

LeoLuosifen committed:
fix: aligned development branch versions of 3 days ago to avoid conflicts.

LeoLuosifen committed:
fix: updated new column names

LeoLuosifen committed:
fix: replaced the string with new column names

LeoLuosifen committed:
fix: renamed the python file of global_element_ids.py

LeoLuosifen committed:
fix: resolved the format to avoid conflict

Commit:
Fix: updated the file global_element_ids.py and summary.py

LeoLuosifen committed:
fix: replaced the string with global constants

LeoLuosifen committed:
fix: resolved the conflict issues

LeoLuosifen committed:
fix: resolved the conflict issues for three md documentations

LeoLuosifen committed:
fix: resolved the conflict issues with global-diffuse-and-normal-solar-radiation-explained.md and the README.md which in sun-and-cloud folder

LeoLuosifen committed:
fix: formatted the code style via using Black in global_column_names.py

FedericoTartarini commented:
@coderabbitai review

FedericoTartarini commented on pages/lib/charts_sun.py:
```diff
@@ -112,7 +113,7 @@ def polar_graph(df, meta, global_local, var, si_ip):
     latitude = float(meta["lat"])
```
these should als be udpated

FedericoTartarini commented on pages/lib/charts_sun.py:
```diff
@@ -196,7 +197,7 @@ def polar_graph(df, meta, global_local, var, si_ip):
         fig.add_trace(
             go.Scatterpolar(
                 r=90 * np.cos(np.radians(90 - solpos["apparent_zenith"])),
```
here as well

FedericoTartarini commented on pages/lib/charts_sun.py:
```diff
@@ -37,22 +38,22 @@ def monthly_solar(epw_df, si_ip):
                     + "Global Horizontal Solar Radiation"
                     + ": %{y:.2f} "
-                    + mapping_dictionary["glob_hor_rad"][si_ip]["unit"]
+                    + mapping_dictionary[ColNames.GLOB_HOR_RAD][si_ip]["unit"]
```
here

FedericoTartarini commented on pages/lib/charts_sun.py:
```diff
@@ -240,7 +241,7 @@ def polar_graph(df, meta, global_local, var, si_ip):
         times = pd.date_range(date, date + pd.Timedelta("24h"), freq="5min", tz=tz)
```
these should also be in a class

FedericoTartarini commented on pages/lib/global_column_names.py:
```diff
@@ -0,0 +1,70 @@
+    OSKYCOVER = "Oskycover"  # Opaque Sky Cover
+    VIS = "Vis"  # Visibility
+    CHEIGHT = "Cheight"  # Cloud Height
+    PWobs = "PWobs"  # Precipitation Observation
```
consistency

FedericoTartarini commented on pages/outdoor.py:
```diff
@@ -212,15 +213,27 @@ def layout():
+        Description of the best weather condition(s).
+    """
     cols = [
         "utci_noSun_Wind_categories",
```
here

FedericoTartarini commented on pages/lib/global_column_names.py:
```diff
@@ -0,0 +1,70 @@
+    AsolOptD = "AsolOptD"  # Aerosol Optical Depth
+    SnowD = "SnowD"  # Snow Depth
+    DaySSnow = "DaySSnow"  # Daily Snow
+    ELEVATION = "elevation"  # Elevation
```
from here down use another class

FedericoTartarini commented on pages/lib/global_column_names.py:
```diff
@@ -0,0 +1,70 @@
+    DAY = "day"  # day
+    HOUR = "hour"  # hour
+    MINUTE = "minute"  # minute
+
```
from enum import Enum

    class ColValues:
        TYPE = 'number'
        NAME = 'name'


    class ColNames(str, Enum):
        # ==================== Time related column ====================
        YEAR = ColValues(
            NAME="year", TYPE="int"  # year
        )
        MONTH = "month"  # month
        DAY = "day"  # day
        HOUR = "hour"  # hour
        MINUTE = "minute"  # minute
    
    col_to_float = []
    for col in ColNames:
        if col.TYPE == 'number':
            col_to_float.append(col.NAME)

Commit:
Fix: updated the ElementIds in file wind

Sallie00 committed:
fix: replaced the string that using Class ColNames in charts_sun.py and charts_summary.py

ziqi0067 committed:
Fix:Update the elements of global_column_names.py and charts_sun.py.

Commit:
Fix: Update the column name in file charts_data_explorer.py

Commit:
Fix: Update the column name in file extract_df.py anad global_column_names.py

ziqi0067 committed:
Fix:Update the elements of template_graphs.py.

Commit:
Fix: Update the column name in file global_scheme.py and psy-chart.py

ziqi0067 committed:
Fix:Update the elements of global_scheme.py, natural_ventilation.py, psy-chart.py, select.py, summary.py.

Commit:
Fix: Update the column name in file global_scheme.py

Sallie00 committed:
Merge branch 'Ziqi' into wlyu

Commit:
Merge remote-tracking branch 'refs/remotes/origin/Ziqi' into ltc

Sallie00 committed:
fix: replaced the string that using Class ColNames in utils.py and summary.py and extract_df.py

Commit:
Merge remote-tracking branch 'refs/remotes/origin/wlyu' into ltc

Commit:
Fix: Update the column name in file global_column_names.py, layout.py and template_graphs.py

LeoLuosifen committed:
fix: replaced the string with the constants

LeoLuosifen committed:
Merge branch 'development' into yluo0664

LeoLuosifen committed:
fix: replaced the string with the constants in outdoor.py

FedericoTartarini commented:
@coderabbit review

FedericoTartarini committed:
style: ruff format

Refactor multiple files to enhance code clarity by adjusting line breaks,
spacing, and indentation. This includes consistent formatting in function
calls and variable assignments across various modules.

FedericoTartarini committed:
fix(layout): Correct label text for yearly chart

Update the label text from "Yearly_chart" to "Yearly Chart" for better
readability and consistency in the user interface.

LeoLuosifen committed:
fix: updated the mistake for using constant

LeoLuosifen commented on pages/summary.py:
```diff
@@ -212,35 +214,35 @@ def update_map(meta):
 
-    site_elevation = float(meta["site_elevation"])
+    site_elevation = float(meta[ColNames.SITE_ELEVATION.SITE_ELEVATION])
     site_elevation = round(site_elevation, 2)
     if si_ip != UnitSystem.SI:
         site_elevation = site_elevation * 3.281
         site_elevation = round(site_elevation, 2)
         elevation = f"Elevation above sea level: {str(site_elevation)} ft"
     else:
-        elevation = f"Elevation above sea level: {meta['site_elevation']} m"
+        elevation = f"Elevation above sea level: {meta[ColNames.SITE_ELEVATION]} m"
     period = ""
```
Sorry for this bug, I'll update it now

LeoLuosifen commented:
@CodeRabbit review

FedericoTartarini committed:
feat(global): Add seasonal wind rose document ID

Introduce a new constant for the seasonal wind rose document ID in
global_element_ids.py and update the wind.py to utilize this ID
for the corresponding button.

FedericoTartarini committed:
Merge remote-tracking branch 'LeoLuosifen/yluo0664' into fork/LeoLuosifen/yluo0664

FedericoTartarini commented:
@LeoLuosifen Thank you for working on this, nice work. However please note the following issues:
* you did not created an ID for the id_button in the title_with_tooltip functions. Please add these to the ElementsIds and commit the changes. For example, in line 61 of the sun.py file.
* always trigger the coderabbit review and address all the comments related to the changes you implemented.
* Resolve conversations, for example, it does not look like you extracted in a class the time in line 241 of the chart_sun. Do not ignore them
* Files were not properly formatted with ruff. I did that.
* Line 221 for the sun.py file should not contain a string, replace with the column name class. Check the project since this issue is also in line 237 and 233 of the same file. This issue is also present in other files.
* pull the latest changes since I did a couple of commits.
* t_rh.py line 95 you should also be using the columns names.
* never reply to comments with messages like "sorry I will fix it" just fix it to keep messages to a minimum

Please review all these issues and make sure that error looks like the above are all fixed so i can do a final review.

Commit:
Fix: Add a new file called global_id_buttons.py

Commit:
Merge branch 'yluo0664' into ltc

LeoLuosifen committed:
Merge branch 'ltc' into yluo0664

LeoLuosifen committed:
fix(layout): fixed the class name to avoid can not load Normalize data chart

Commit:
Merge branch 'yluo0664' into ltc

Sallie00 committed:
fix: replaced the string that using Class IdButtons in explorer.py, natural_ventilation.py, psy-chart.py, summary.py, sun.py, wind.py, t-rh.py and outdoor.py

Commit:
Merge branch 'wlyu' into ltc

Commit:
Fix: Updated ids of files global_element_ids.py,layout.py,main.py,select.py

LeoLuosifen committed:
fix: fixed code sanitisation
- add global tab names python file to store tab name value
- replaced the tab_name via using global_tab_names.py
- removed unused import: `enum.Enum`
- formatted the code style via running ruff and pre-commit

LeoLuosifen commented:
@CodeRabbit review

LeoLuosifen committed:
fix: fixed the id_button aligned with IdButtons.* and aligned var/file_var to constants.

FedericoTartarini committed:
fix(data): Replace string references with ColNames for wind speed

Update instances of "wind_speed" to ColNames.WIND_SPEED in multiple files
to improve code consistency and maintainability.

FedericoTartarini committed:
fix(layout): Correct capitalization of yearly chart label

Update the label for the yearly chart to use consistent capitalization
for improved readability and consistency in the UI.
